### DISCUSSION TITLE
CSS selector - Images

### DISCUSSION THREAD

**Discussion by LukeG20** [FreshRSS/FreshRSS#4958] [Q&A]
Is there a way for the Article CSS selector to also pull back pictures? 

Not sure if its a bug but it pulls back the wrong URL for example on ANN it replaced 

https://cdn.animenewsnetwork.com/thumbnails/fit300x1000/cms/preview-guide/190198/sub_csm_01_336.png.jpg with
https://www.animenewsnetwork.com/img/spacer.gif

**Comment by Frenzie** [comment]
Could you provide a bit more info? (FreshRSS version, feed URL, CSS selector)

**Comment by LukeG20** [comment]
@Frenzie thanks for the reply! 
Version: 1.20.2
feed URL: https://www.animenewsnetwork.com/preview-guide/rss.xml?ann-edition=w
CSS selector: #maincontent

The feed comes out looking like this:
![image](https://user-images.githubusercontent.com/108997932/210006576-14320c88-cf47-43f6-9740-02ad07121bb7.png)

All the images are missing, looking in the element inspector it seems to be replacing links for some reason.

Example of one such link:
In FreshRSS:
![image](https://user-images.githubusercontent.com/108997932/210006386-bdb2dff3-30d0-492c-94d2-f1be01e3e701.png)


On original website
![image](https://user-images.githubusercontent.com/108997932/210006427-36f9ba7c-e6f4-4008-8541-911b9a1d0282.png)

I'm not sure if this is just a limitation of CSS selector and expected behavior or I'm doing something wrong.

**Reply by Frenzie** [reply]
The HTML source and the DOM are often very similar, but that's where the difference stems from. (As can be intuited from your screenshot.)

The HTML from the page is: `<img src="/img/spacer.gif" width="1000" height="389" data-src="/thumbnails/max1000x1500/cms/preview-guide/186807/summerbanner_2022.jpg" class="lazyload">`. That'll be replaced later by a script.

We actually already have a workaround for this particular scenario in regular feeds, see https://github.com/FreshRSS/FreshRSS/issues/3823#issuecomment-913212244

**Reply by LukeG20** [reply]
So there's no way around it when using CSS selector then? I just tried the method detailed in #3823, which didn't work in this instance.

Guessing the 44 in "i/?a=normal&get=f_44" is the feed id?

**Reply by Frenzie** [reply]
Not using CSS, no. Swapping attributes is a job for JS (or PHP).

You might be able to use solely CSS in this particular case by using the `data-src` attribute as a `background-image`, if you want a possibly fun challenge. :-)
